app/uk/gov/hmrc/exports/migrations/changelogs/submission/AddActionFieldsForAmend.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.migrations.changelogs.submission
+
+import com.mongodb.client.MongoDatabase
+import com.mongodb.client.model.Updates.{combine, set}
+import org.mongodb.scala.model.{Filters, UpdateOptions}
+import org.mongodb.scala.model.Filters._
+import play.api.Logging
+import uk.gov.hmrc.exports.migrations.changelogs.{MigrationDefinition, MigrationInformation}
+import uk.gov.hmrc.exports.models.declaration.submissions._
+
+import java.util
+import scala.jdk.CollectionConverters._
+
+class AddActionFieldsForAmend extends MigrationDefinition with Logging {
+
+  override val migrationInformation: MigrationInformation =
+    MigrationInformation(
+      id = s"CEDS-4446 Add new fields to the Action model for amended Declarations.",
+      order = 18,
+      author = "Darryl Murphy",
+      runAlways = true
+    )
+  private val uuid = "uuid"
+  private val latestDecId = "latestDecId"
+  private val latestVersionNo = "latestVersionNo"
+
+  override def migrationFunction(db: MongoDatabase): Unit = {
+
+    logger.info(s"Applying '${migrationInformation.id}' db migration...")
+
+    val submissionCollection = db.getCollection("submissions")
+
+    submissionCollection.find(Filters.and(exists("versionNo", false), exists("decId", false))).asScala.map { document =>
```

Comment:
This filter does not work as intended. I think it is looking for fields at the submission level (that are not there), where it should be looking for fields at the action level. 
I tried running this migration locally and it overwrote existing action records that already had these fields defined.